doc/contribute/index.rst

```diff
@@ -356,7 +356,12 @@ following exceptions:
 * Use C89-style single line comments, ``/*  */``. The C99-style single line
   comment, ``//``, is not allowed.
 * Use ``/**  */`` for doxygen comments that need to appear in the documentation.
+* Conditional compilation:
 
+  * Do not conditionally compile function declarations in header files.
+  * Do not conditionally compile structure declarations in header files.
+  * You may conditionally exclude fields within structure definitions to avoid
+    wasting memory when the feature they support is not enabled.
```

Comment:
This will need clarification if/when we get `DOXYGEN` added to `doc/zephyr.doxyfile.in` replacing `DT_DOXYGEN`, because the guideline should then explain *how* to conditionally exclude those fields.

Reply:
the DOXYGEN macro might work for some cases, but I really do not want to see code where Kconfig variables are set to values in headers files to satisfy doxygen, this is going to be a lot of noise and confusing above all.